python/cudf/cudf/utils/utils.py

```diff
@@ -119,31 +132,39 @@ def buffers_from_pyarrow(pa_arr, dtype=None):
 
     buffers = pa_arr.buffers()
 
-    if buffers[0]:
+    if pa_arr.null_count:
         mask_dev_array = make_mask(len(pa_arr))
         arrow_dev_array = rmm.to_device(np.array(buffers[0]).view("int8"))
         copy_array(arrow_dev_array, mask_dev_array)
         pamask = Buffer(mask_dev_array)
     else:
         pamask = None
 
+    offset = pa_arr.offset
+    size = pa_arr.offset + len(pa_arr)
```

Comment:
Is the offset in number of elements or the number of bytes?

Reply:
It is in the number of elements - I assume that pyarrow is the same?

Reply:
Yup: https://github.com/apache/arrow/blob/master/python/pyarrow/array.pxi#L753